R/rintrinio.R

```diff
@@ -36,8 +36,85 @@ gather_financial_statement_time_series <- function(api_key, ticker, statement, y
 #' @examples
 #' gather_financial_statement_company_compare(api_key, ['AAPL', 'CSCO'], 'income_statement', '2019', 'Q1')
 
-gather_financial_statement_company_compare <- function(api_key, ticker, statement, year, period) {
-  tibble()
+gather_financial_statement_company_compare <- function(api_key, ticker, statement, year, period){
+  
+  #check if the api_key is a string
+  if (typeof(api_key) != "character"){
+    stop("the api_key must be a string")
+  }
+  
+  #check if the ticker is a string
+  if (typeof(ticker) != "character") {
+    stop("the ticker must be a string")
+  }
+  
+  #check if the statement is a string
+  if (typeof(statement) != "character") {
+    stop("the statement must be a string")
+  }
+  
+  #check if the year is a string
+  if (typeof(year) != "character") {
+    stop("the year must be a string")
+  }
+  
+  #check if the period is a string
+  if (typeof(period) != "character") {
+    stop("the period must be a string")
+  }
+  
+  
+  client <- IntrinioSDK::ApiClient$new()
+  
+  # Configure API key authorization: ApiKeyAuth
+  client$configuration$apiKey <- key
+  
+  # Setup API with client
+  FundamentalsApi <- IntrinioSDK::FundamentalsApi$new(client)
+  
+  #final dataframe
+  result = c()
+  
+  for (comp in seq(length(ticker))){
+    
+    #set the id
+    id <- paste(ticker[comp], statement, year, period, sep='-')
+    
+    response <- FundamentalsApi$get_fundamental_standardized_financials(id)
```

Comment:
Similar to Roc's and my functions, can we make it throw an error for an invalid API key?